src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CreateClient.cs

```diff
@@ -0,0 +1,127 @@
+        public static void InvalidPipeDirection_Throws_ArgumentOutOfRangeException(PipeDirection direction)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("direction", () => new NamedPipeClientStream(".", "client1", direction));
+            Assert.Throws<ArgumentOutOfRangeException>("direction", () => new NamedPipeClientStream(".", "client1", direction, PipeOptions.None));
+            Assert.Throws<ArgumentOutOfRangeException>("direction", () => new NamedPipeClientStream(".", "client1", direction, PipeOptions.None, Security.Principal.TokenImpersonationLevel.None));
+        }
+
+        [Theory]
+        [InlineData(PipeDirection.In)]
+        [InlineData(PipeDirection.InOut)]
+        [InlineData(PipeDirection.Out)]
+        public static void InvalidPipeOptions_Throws_ArgumentOutOfRangeException(PipeDirection direction)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("options", () => new NamedPipeClientStream(".", "client1", direction, (PipeOptions)255));
+            Assert.Throws<ArgumentOutOfRangeException>("options", () => new NamedPipeClientStream(".", "client1", direction, (PipeOptions)255, Security.Principal.TokenImpersonationLevel.None));
+        }
+
+        [Theory]
+        [InlineData(PipeDirection.In)]
+        [InlineData(PipeDirection.InOut)]
+        [InlineData(PipeDirection.Out)]
+        public static void InvalidImpersonationLevel_Throws_ArgumentOutOfRangeException(PipeDirection direction)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("impersonationLevel", () => new NamedPipeClientStream(".", "client1", direction, PipeOptions.None, (System.Security.Principal.TokenImpersonationLevel)999));
+        }
+
+        [Theory]
+        [InlineData(PipeDirection.In)]
+        [InlineData(PipeDirection.InOut)]
+        [InlineData(PipeDirection.Out)]
+        public static void NullHandle_Throws_ArgumentNullException(PipeDirection direction)
+        {
+            Assert.Throws<ArgumentNullException>("safePipeHandle", () => new NamedPipeClientStream(direction, false, true, null));
+        }
+
+        [Theory]
+        [InlineData(PipeDirection.In)]
+        [InlineData(PipeDirection.InOut)]
+        [InlineData(PipeDirection.Out)]
+        public static void InvalidHandle_Throws_ArgumentException(PipeDirection direction)
+        {
+            SafePipeHandle pipeHandle = new SafePipeHandle(new IntPtr(-1), true);
+            Assert.Throws<ArgumentException>("safePipeHandle", () => new NamedPipeClientStream(direction, false, true, pipeHandle));
+        }
+
+        [Theory]
+        [InlineData(PipeDirection.In)]
+        [InlineData(PipeDirection.InOut)]
+        [InlineData(PipeDirection.Out)]
+        public static void BadHandleKind_Throws_IOException(PipeDirection direction)
+        {
+            using (FileStream fs = new FileStream(Guid.NewGuid().ToString(), FileMode.Create, FileAccess.Write, FileShare.None, 8, FileOptions.DeleteOnClose))
```

Comment:
Can we use the same temporary file mechanism we've used in the FileSystem and MMF suites?